newIDE/app/src/ObjectsRendering/Renderers/RenderedCustomObjectInstance.js

```diff
@@ -142,21 +194,58 @@ export default class RenderedCustomObjectInstance extends Rendered3DInstance
   ): RenderedInstance | Rendered3DInstance => {
     let renderedInstance = this.renderedInstances.get(instance.ptr);
     if (!renderedInstance) {
+      // No renderer associated yet, the instance must have been just created!...
+      let childObjectConfiguration = null;
+      const variant = this.getVariant();
+      if (variant) {
+        const childObjects = variant.getObjects();
+        if (childObjects.hasObjectNamed(instance.getObjectName())) {
+          const childObject = childObjects.getObject(instance.getObjectName());
+          childObjectConfiguration = childObject.getConfiguration();
+        }
+      }
+      // Apply property mapping rules on the child instance.
+      const childPropertyOverridings = new Map<string, string>();
       const customObjectConfiguration = gd.asCustomObjectConfiguration(
         this._associatedObjectConfiguration
       );
-      //No renderer associated yet, the instance must have been just created!...
-      const childObjectConfiguration = customObjectConfiguration.getChildObjectConfiguration(
-        instance.getObjectName()
-      );
+      for (const propertyMappingRule of this._propertyMappingRules) {
+        if (propertyMappingRule.targetChild !== instance.getObjectName()) {
+          continue;
+        }
+        const sourceValue = this._propertyOverridings.has(
+          propertyMappingRule.sourceProperty
+        )
+          ? this._propertyOverridings.get(propertyMappingRule.sourceProperty)
+          : customObjectConfiguration
```

Comment:
Calling getProperties multiple times can be resource intensive, consider doing it once if you think that's better and still safe.